### DISCUSSION TITLE
Can I make obsidian like desktop app with dioxus

### DISCUSSION THREAD

**Discussion by blueray453** [DioxusLabs/dioxus#2553] [Q&A] [answered]
I am mainly interested in graph visualization. Can I create concept map (similar to mind map but the arrows will have label). 

The nodes should be moveable by dragging.

I would also like to click and right click on the nodes and relationships (label of the arrows) to have a action or menu to run different functions. 

If it is not possible then some sort of webview that let me use d3js will also work.

Thanks.

**Comment by ealmloff** [comment, marked as answer]
Yes, but you will need to build a lot of that logic yourself or copy it from existing projects. I don't know of any draggable nodes library for dioxus. [Floneum](https://github.com/floneum/floneum/tree/main/floneum/floneum) uses dioxus desktop. It has draggable nodes with a mix of html and svgs. Feel free to pull out some of the logic for dragging around svgs

**Reply by blueray453** [reply]
Thank you very much for the insight.
